src/MyEventEmitter.js

```diff
@@ -1,14 +1,78 @@
 'use strict';
 
 class MyEventEmitter {
-  on() {}
-  once() {}
-  off() {}
-  emit() {}
-  prependListener() {}
-  prependOnceListener() {}
-  removeAllListeners() {}
-  listenerCount() {}
+  listeners = {};
+
+  ensureEventExists(eventName) {
+    if (!this.listeners[eventName]) {
+      this.listeners[eventName] = [];
+    }
+  }
+
+  on(eventName, callback) {
+    this.ensureEventExists(eventName);
+
+    this.listeners[eventName].push({ callback });
+  }
+
+  once(eventName, callback) {
+    this.ensureEventExists(eventName);
+
+    this.listeners[eventName].push({ callback, once: true });
+  }
+
+  off(eventName, callback) {
+    if (!this.listeners[eventName]) {
+      return;
+    }
+
+    this.listeners[eventName] = this.listeners[eventName].filter(
+      (listener) => listener.callback !== callback,
+    );
+  }
+
+  emit(eventName, ...arg) {
+    const listenersForEvent = this.listeners[eventName];
+
+    if (!listenersForEvent) {
+      return;
+    }
+
+    for (let i = 0; i < listenersForEvent.length; i++) {
+      const currentListener = listenersForEvent[i];
+
+      currentListener.callback(...arg);
+
+      if (currentListener.once) {
+        listenersForEvent.splice(i, 1);
+        i--;
```

Comment:
The logic for removing a listener and decrementing the index is correct to avoid skipping the next listener after a 'once' listener is removed. However, ensure that this logic is carefully maintained if the code is modified, as it can lead to potential issues if not handled properly.